src/mono/mono/metadata/marshal-ilgen.c

```diff
@@ -609,7 +620,11 @@ conv_to_icall (MonoMarshalConv conv, int *ind_store_type)
 		return MONO_JIT_ICALL_ves_icall_mono_string_from_utf16;
 	case MONO_MARSHAL_CONV_LPTSTR_STR:
 		*ind_store_type = CEE_STIND_REF;
+#ifdef TARGET_WIN32
+		return MONO_JIT_ICALL_ves_icall_mono_string_from_utf16;
+#else
```

Comment:
Not sure about this.

Reply:
I'll move this into a separate PR, but I'm pretty sure it's wrong as-is. Right now, marshaling LPTSTR and back will have a broken conversion, see https://github.com/dotnet/runtime/pull/35190/files#diff-b2e5b24920b37f04d29374c6ec4856c4L467. Given the docs specify it's a platform-dependent string, this looks right to me.

From what I can tell AnsiBSTR, LPSTR, LPTSTR, and TBSTR are all partially wrong or unimplemented in netcore. AnsiBSTR only goes one way, LPSTR is treated as a UTF8STR for legacy reasons (not applicable in netcore), LPTSTR has this issue so I think I could cook up a test case that would cause a crash, and TBSTR only goes one way.

Sadly, only AnsiBSTR, BSTR, and LPSTR are covered by the CoreCLR tests right now, but I think I should be able to reuse the existing tests with the other types as well. There are a lot of interop-related test failures, but hopefully that will clear up any with the strings themselves.

Reply:
Opened an issue: https://github.com/dotnet/runtime/issues/35196